plugins/woocommerce/client/legacy/js/frontend/ajax-add-to-cart.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * This is an interim solution for handling AJAX add-to-cart functionality,
+ * allowing features like the mini-cart drawer to open when products are added.
+ * This will eventually be replaced by the blockified Add to Cart implementation
+ * tracked in https://github.com/woocommerce/woocommerce/issues/53014
+ *
+ * The script:
+ * 1. Captures the add-to-cart button click
+ * 2. Serializes the product form data
+ * 3. Submits the data via AJAX
+ * 4. Triggers appropriate events for cart updates
+ */
+( function ( $ ) {
```

Comment:
@nielslange I realize this is POC, but just stating: please do not add any new jQuery code to the codebase even in legacy frontend. We can implement everything here with plain JS.

The only exception for writing jQuery would be to maintain some existing script that would be unreasonable/untenable to refactor to plain JS. See pcShBQ-1S4-p2

Reply:
Thanks for pointing this out, @samueljseay, and thanks for sharing the link. In case we decide to proceed in the direction this POC points to, I'll make sure to implement the solution in plain JS.